src/radical/entk/execman/rp/task_processor.py

```diff
@@ -473,9 +473,9 @@ def create_td_from_task(task, placeholders, task_hash_table, pkl_path, sid,
         td.stage_on_error = task.stage_on_error
 
         if task.parent_pipeline['uid']:
-            td.tag = resolve_tags(task=task,
+            td.tags = {'colocate': resolve_tags(task=task,
```

Comment:
I'd like to propose that method `resolve_tags` returns the actual structure of `tags` attribute, since it is responsible to resolve it
```python
def resolve_tags(...
    ...
    return {'colocate': task.uid}
```

Reply:
This is now resolved. @mtitov can you check if this is okay with you?